api/services/auth.go

```diff
@@ -283,6 +288,112 @@ func (s *service) AuthGetToken(ctx context.Context, id string, mfa bool) (*model
 	}, nil
 }
 
+func (s *service) AuthCustomToken(ctx context.Context, id, nameToken string, expiredAt int) (map[string]interface{}, error) {
+	user, _, err := s.store.UserGetByID(ctx, id, false)
+	if err != nil {
+		return nil, NewErrUserNotFound(id, err)
+	}
+
+	namespace, _ := s.store.NamespaceGetFirst(ctx, user.ID)
+
+	var role string
+	var tenant string
+	if namespace != nil {
+		tenant = namespace.TenantID
+
+		for _, member := range namespace.Members {
+			if member.ID == user.ID {
+				role = member.Role
+
+				break
+			}
+		}
+	}
+
+	expiredTime := time.Now()
+
+	switch expiredAt {
+	case 30, 60, 90:
+		expiredTime = expiredTime.Add(time.Hour * 24 * time.Duration(expiredAt))
+	case 1:
+		expiredTime = expiredTime.AddDate(1, 0, 0)
+	case -1:
+		expiredTime = expiredTime.AddDate(100, 0, 0)
+	default:
+		return nil, errors.New("experid date to custom token is invalid")
+	}
+
+	mfa, err := s.AuthMFA(ctx, user.ID)
+	if err != nil {
+		return nil, NewErrUserNotFound(id, err)
+	}
+
+	token := jwt.NewWithClaims(jwt.SigningMethodRS256, models.UserAuthClaims{
+		Username: user.Username,
+		Admin:    true,
+		Tenant:   tenant,
+		Role:     role,
+		ID:       user.ID,
+		AuthClaims: models.AuthClaims{
+			Claims: "user",
+		},
+		MFA: models.MFA{
+			Enable:   mfa,
+			Validate: mfa,
+		},
+		RegisteredClaims: jwt.RegisteredClaims{
+			ExpiresAt: jwt.NewNumericDate(expiredTime),
+		},
+	})
```

Comment:
We have a [jwttoken](https://github.com/shellhub-io/shellhub/blob/master/pkg/api/jwttoken/jwttoken.go), which is an abstraction layer around jwt. Perhaps you could utilize it? You can see an example [here](https://github.com/shellhub-io/shellhub/blob/master/api/services/auth.go#L178-L196).

Reply:
changed to your package to create a jwtToken